src/engine.ml

```diff
@@ -101,6 +101,21 @@ let hmac_and_out protocol key { hmac_algorithm; my_hmac; _ } (p : Packet.pkt) =
   let p' = Packet.with_header { header with Packet.hmac } p in
   Packet.encode protocol (key, p')
 
+let encrypt_and_out protocol key { my_key; my_hmac; _ } (p : Packet.pkt) =
+  let to_be_signed = Packet.Tls_crypt.to_be_signed key p in
+  let hmac = Mirage_crypto.Hash.SHA256.hmac ~key:my_hmac to_be_signed in
+  let iv = Cstruct.sub hmac 0 16 in
```

Comment:
is the 16 the block size of AES that is used? if yes, should we use `Mirage_crypto.AES.CTR.block_size` instead?

Reply:
If I remember correctly in the documentation they specify it as the first half of the hmac, or the 128 most significant bits (without specifying endianness).

Reply:
fine then